molgenis-security/src/main/java/org/molgenis/security/permission/MolgenisPermissionController.java

```diff
@@ -37,4 +64,53 @@ public boolean hasWritePermission(@PathVariable("entityTypeId") String entityTyp
 		return molgenisPermissionService
 				.hasPermissionOnEntityType(entityTypeId, org.molgenis.security.core.Permission.WRITE);
 	}
+
+	@RequestMapping(value = "/sid", method = GET)
+	@ResponseBody
+	public Collection<SecurityId> getSids()
+	{
+		return dataService.findAll(ROLE, Role.class)
+						  .map(role -> SecurityId.createForAuthority(role.getLabel()))
+						  .collect(toList());
+	}
+
+	@RequestMapping(value = "/acl", method = PUT)
+	@ResponseStatus(code = HttpStatus.NO_CONTENT)
+	public void save(@RequestBody EntityAcl acl)
+	{
+		entityAclManager.updateAcl(acl);
+	}
+
+	/**
+	 * Retrieves the ACLs for entities of a specific EntityType
+	 *
+	 * @param entityTypeId entity type ID
+	 * @param filter       search string to filter on
+	 * @param pageSize     number of ACLs to retrieve
+	 * @return the retrieved {@link EntityAcl}s
+	 */
+	@RequestMapping(value = "/acls", produces = "application/json", method = GET)
+	@ResponseBody
+	public List<Map> entityAcls(@RequestParam String entityTypeId, @RequestParam(required = false) String filter,
+			@RequestParam(defaultValue = "10") int pageSize)
+	{
+		EntityType entityType = dataService.getEntityType(entityTypeId);
+		String languageCode = languageService.getCurrentUserLanguageCode();
+		String labelAttribute = entityType.getLabelAttribute(languageCode).getName();
+
+		QueryImpl<Entity> query = new QueryImpl<>();
+		if (isNotBlank(filter))
+		{
+			query.addRule(new QueryRule(SEARCH, filter.trim()));
```

Comment:
As mentioned in person by @fdlk: the indexing is broken on the security branch so SEARCH will not work as expected.